lib/browser/rpc-server.js

```diff
@@ -309,8 +309,19 @@ handleRemoteCommand('ELECTRON_BROWSER_REQUIRE', function (event, contextId, modu
   return valueToMeta(event.sender, contextId, customEvent.returnValue)
 })
 
-handleRemoteCommand('ELECTRON_BROWSER_GET_BUILTIN', function (event, contextId, module) {
-  return valueToMeta(event.sender, contextId, electron[module])
+handleRemoteCommand('ELECTRON_BROWSER_GET_BUILTIN', function (event, contextId, moduleName) {
+  const customEvent = eventBinding.createWithSender(event.sender)
+  event.sender.emit('remote-get-builtin', customEvent, moduleName)
+
+  if (customEvent.defaultPrevented) {
```

Comment:
it's surprising to me that you have to `preventDefault()` _as well as_ setting `returnValue` to return a custom value. My expectation from reading the docs is that this would work:

```js
webContents.on('remote-get-builtin', (e, mod) => {
  if (mod === 'something') {
    e.returnValue = someWrapper()
  } else {
    e.preventDefault()
  }
})
```

I can certainly imagine it taking some head-scratching to figure out that you have to call `preventDefault` in order for your custom return value to be respected.

Reply:
The other events that we already have in master and 4.0 behave like that. I can change it in a separate PR for all events consistently.

Reply:
yeah :( sad that it didn't get caught earlier.

is there a web API that behaves like the current way that you were following? perhaps i'm missing context